### DISCUSSION TITLE
SQS triggered lambda (from issue 7069)

### DISCUSSION THREAD

**Discussion by brunetto** [pulumi/pulumi#8904] [Pulumi in Practice]
Hi, 
I'm trying to create a lambda triggered by SQS. 
Following the docs linked in the issue https://github.com/pulumi/docs/issues/7069 (see https://www.pulumi.com/registry/packages/aws/api-docs/lambda/eventsourcemapping/#sqs) I was able (with some modifications) to have something that MAY work:

```go
func QueueTriggersFunction(ctx *pulumi.Context, triggerName string, queue *sqs.Queue, fn *lambda.Function) error {
	_, err := lambda.NewEventSourceMapping(ctx, "sqs_trigger_lambda", &lambda.EventSourceMappingArgs{
		EventSourceArn: pulumi.StringOutput(queue.Arn),
		FunctionName:   pulumi.StringOutput(fn.Arn),
	})
	if err != nil {
		return errors.Wrap(err, "fail to integrate lambda with sqs")
	}

	return nil
}
``` 

However, running `pulumi up` results in the following error:

```
        * error creating Lambda Event Source Mapping (arn:aws:sqs:...): InvalidParameterValueException: The provided execution role does not have permissions to call ReceiveMessage on SQS
```

Is there a _complete_ example of what it's needed to have it working? 
I'm sorry if there's already a doc about it, but I was unable to find it.

**Comment by nimbinatus** [comment]
I don't know of a complete example, but that sounds like something's up with the role creation elsewhere in the code rather than with that function. I think the default role that lambdas get doesn't get access to SQS.

If you're defining the role policy in your code with something like [this snippet](https://github.com/pulumi/examples/blob/master/aws-go-lambda/main.go#L29-L44), you'll want to add the roles for SQS access:
```
sqsPolicy, err := iam.NewRolePolicy(ctx, "lambda-sqs-policy", &iam.RolePolicyArgs{
	Role: role.Name,
	Policy: pulumi.String(`{
"Version": "2012-10-17",
"Statement": [{
    "Effect": "Allow",
    "Action": [
      "sqs:ReceiveMessage",
      "sqs:DeleteMessage",
      "sqs:GetQueueAttributes"
    ],
    "Resource": "arn:aws:sqs:*:*:*"
}]
}`),
})
```
